File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-0.py
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        mapping = {
            "2" : ["a", "b", "c"],
            "3" : ["d", "e", "f"],
            "4" : ["g", "h", "i"],
            "5" : ["j", "k", "l"],
            "6" : ["m", "n", "o"],
            "7" : ["p", "q", "r","s"],
            "8" : ["t", "u", "v"],
            "9" : ["w", "x", "y", "z"]
        }

        res = []
        curr_str = ""
        
        def backtrack(i, curr_str):

            if len(curr_str) == len(digits):
                res.append(curr_str)
                return


            for char in mapping[digits[i]]:
                backtrack(i+1, curr_str+char)


        if len(digits):
            backtrack(0, curr_str)

        return res

        

File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string res = "";

        for(int i=0; i <size(strs[0]); i++){
            for(int s=0; s<size(strs);s++){
                if (i == strs[s].size() || strs[0][i] != strs[s][i]){
                    return res;
                }
            }
            res += strs[0][i];
        }
        return res;
    }
};